l10n_jp_post_code_propose_partner_address/models/res_partner.py

```diff
@@ -0,0 +1,69 @@
+# Copyright 2019 Quartile Limited
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+import jaconv
+import json
+import urllib
+
+from odoo import models, fields, api, _
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    country_id = fields.Many2one(
+        default=lambda self: self.env.ref('base.jp')
+    )
+
+    @api.onchange('zip')
```

Comment:
Is it by intention that we use onchange here (i.e. we don't want to trigger address update for some cases - when we import partner data, for example)?

It may also be good to place a button to trigger the address update to give users some control over when the update should take place.